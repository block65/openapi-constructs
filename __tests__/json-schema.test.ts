/* eslint-disable no-new */
import { Ajv } from 'ajv';
import { test, expect, describe } from 'vitest';
import { exampleApi } from './fixtures/apis/example.js';
import { noteTakingApi } from './fixtures/apis/note-taking.js';

describe('Example', () => {
  const jsonSchema = exampleApi.synthJsonSchema();

  test('JSON Schema snapshot', async () => {
    expect(jsonSchema).toMatchSnapshot();
  });

  test('JSON Schema AJV validate', async () => {
    const ajv = new Ajv();
    expect(ajv.validateSchema(jsonSchema)).toBeTruthy();
  });
});

describe('Note Taking', () => {
  const jsonSchema = noteTakingApi.synthJsonSchema();

  test('JSON Schema snapshot', async () => {
    expect(jsonSchema).toMatchSnapshot();
  });

  test('JSON Schema AJV validate', async () => {
    const ajv = new Ajv();
    expect(ajv.validateSchema(jsonSchema)).toBeTruthy();
  });
});
