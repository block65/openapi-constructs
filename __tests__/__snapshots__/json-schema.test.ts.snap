// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Example > JSON Schema snapshot 1`] = `
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "definitions": {
    "Address": {
      "additionalProperties": false,
      "properties": {
        "postcode": {
          "format": "int32",
          "maximum": 9999,
          "minimum": 1000,
          "type": "integer",
        },
      },
      "required": [
        "name",
      ],
      "type": "object",
    },
    "CreateUserRequest": {
      "additionalProperties": false,
      "properties": {
        "address": {
          "$ref": "#/components/schemas/Address",
        },
        "age": {
          "anyOf": [
            {
              "format": "int32",
              "minimum": 0,
              "type": "integer",
            },
            {
              "type": "null",
            },
          ],
        },
        "name": {
          "type": "string",
        },
        "userId": {
          "$ref": "#/components/schemas/Id",
        },
      },
      "required": [
        "name",
      ],
      "type": "object",
    },
    "Id": {
      "type": "string",
    },
    "UpdateUserRequest": {
      "additionalProperties": false,
      "minProperties": 1,
      "properties": {
        "address": {
          "$ref": "#/components/schemas/Address",
        },
        "age": {
          "format": "int32",
          "minimum": 0,
          "type": "integer",
        },
      },
      "type": "object",
    },
    "User": {
      "additionalProperties": false,
      "properties": {
        "address": {
          "$ref": "#/components/schemas/Address",
        },
        "age": {
          "anyOf": [
            {
              "format": "int32",
              "minimum": 0,
              "type": "integer",
            },
            {
              "type": "null",
            },
          ],
        },
        "name": {
          "type": "string",
        },
        "userId": {
          "$ref": "#/components/schemas/Id",
        },
      },
      "required": [
        "name",
      ],
      "type": "object",
    },
    "Users": {
      "items": {
        "$ref": "#/components/schemas/User",
      },
      "type": "array",
      "uniqueItems": true,
    },
  },
}
`;

exports[`Note Taking > JSON Schema snapshot 1`] = `
{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "definitions": {
    "Address": {
      "additionalProperties": false,
      "properties": {
        "postcode": {
          "format": "int32",
          "maximum": 9999,
          "minimum": 1000,
          "type": "integer",
        },
      },
      "required": [
        "name",
      ],
      "type": "object",
    },
    "CreateNoteRequest": {
      "additionalProperties": false,
      "properties": {
        "content": {
          "type": "string",
        },
        "labels": {
          "items": {
            "maxLength": 256,
            "minLength": 1,
            "type": "string",
          },
          "type": "array",
        },
        "title": {
          "type": "string",
        },
      },
      "required": [
        "title",
        "content",
      ],
      "type": "object",
    },
    "CreateUserRequest": {
      "additionalProperties": false,
      "properties": {
        "address": {
          "$ref": "#/components/schemas/Address",
        },
        "age": {
          "format": "int32",
          "minimum": 0,
          "type": "integer",
        },
        "name": {
          "type": "string",
        },
        "userId": {
          "$ref": "#/components/schemas/Id",
        },
      },
      "required": [
        "name",
      ],
      "type": "object",
    },
    "Id": {
      "maxLength": 6,
      "minLength": 6,
      "type": "string",
    },
    "Note": {
      "additionalProperties": false,
      "properties": {
        "content": {
          "type": "string",
        },
        "labels": {
          "items": {
            "maxLength": 256,
            "minLength": 1,
            "type": "string",
          },
          "type": "array",
        },
        "title": {
          "type": "string",
        },
      },
      "required": [
        "title",
        "content",
      ],
      "type": "object",
    },
    "Notes": {
      "items": {
        "$ref": "#/components/schemas/Note",
      },
      "type": "array",
      "uniqueItems": true,
    },
    "UpdateNoteRequest": {
      "additionalProperties": false,
      "minProperties": 1,
      "properties": {
        "content": {
          "type": "string",
        },
        "labels": {
          "items": {
            "maxLength": 256,
            "minLength": 1,
            "type": "string",
          },
          "type": "array",
        },
        "title": {
          "type": "string",
        },
      },
      "type": "object",
    },
    "UpdateUserRequest": {
      "additionalProperties": false,
      "minProperties": 1,
      "properties": {
        "address": {
          "$ref": "#/components/schemas/Address",
        },
        "age": {
          "format": "int32",
          "minimum": 0,
          "type": "integer",
        },
      },
      "type": "object",
    },
    "User": {
      "additionalProperties": false,
      "properties": {
        "address": {
          "$ref": "#/components/schemas/Address",
        },
        "age": {
          "format": "int32",
          "minimum": 0,
          "type": "integer",
        },
        "name": {
          "type": "string",
        },
        "userId": {
          "$ref": "#/components/schemas/Id",
        },
      },
      "required": [
        "name",
      ],
      "type": "object",
    },
    "Users": {
      "items": {
        "$ref": "#/components/schemas/User",
      },
      "type": "array",
      "uniqueItems": true,
    },
  },
}
`;
